app/lib/data.ts

```diff
@@ -136,7 +158,12 @@ export async function fetchPages(query: string, selectedAuthors: string[]) {
     const totalPages = Math.ceil(Number(count.rows[0].count) / ITEMS_PER_PAGE);
     return totalPages;
   } catch (error) {
-    console.error("Database Error:", error);
-    throw new Error("Failed to fetch total number of books.");
+    if (isTableMissing(error)) {
+      console.error("Database Error - Table missing:", error);
+      return 0;
+    } else {
+      console.error("Database Error:", error);
+      throw new Error("Failed to fetch total number of books.");
+    }
```

Comment:
This same block of code is repeated 4 times in this file. Extract it into a function?

Reply:
Made this change, but it causes TS errors elsewhere in the app since it returns two diff types. Ran out of ideas on how to fix these new errors